src/rules/utils.ts

```diff
@@ -0,0 +1,685 @@
+> extends KnownMemberExpression<PropertyName> {
+  object: ExpectCall | ExpectMember;
+  parent: Parent;
+}
+
+export const isExpectMember = <
+  Name extends ExpectPropertyName = ExpectPropertyName
+>(
+  node: TSESTree.Node,
+  name?: Name,
+): node is ExpectMember<Name> =>
+  node.type === AST_NODE_TYPES.MemberExpression &&
+  isSupportedAccessor(node.property) &&
+  (name === undefined || getAccessorValue(node.property) === name);
+
+/**
+ * Represents all the jest matchers.
+ */
+export type MatcherName = string /* & not ModifierName */;
+export type ExpectPropertyName = ModifierName | MatcherName;
+
+export type ParsedEqualityMatcherCall<
+  Argument extends TSESTree.Expression = TSESTree.Expression
+> = Omit<ParsedExpectMatcher<EqualityMatcher>, 'arguments'> & {
+  // todo: probs should also type node parent as CallExpression
+  arguments: [Argument];
+};
+
+export const isParsedEqualityMatcherCall = (
+  matcher: ParsedExpectMatcher,
+): matcher is ParsedEqualityMatcherCall =>
+  EqualityMatcher.hasOwnProperty(matcher.name) &&
+  matcher.arguments !== null &&
+  matcher.arguments.length === 1;
+
+export enum EqualityMatcher {
+  toBe = 'toBe',
+  toEqual = 'toEqual',
+}
+
+export enum ModifierName {
+  not = 'not',
+  rejects = 'rejects',
+  resolves = 'resolves',
+}
+
+interface ParsedExpectMember<
+  Name extends ExpectPropertyName = ExpectPropertyName,
+  Node extends ExpectMember<Name> = ExpectMember<Name>
+> {
+  name: Name;
+  node: Node;
+}
+
+export interface ParsedExpectMatcher<
+  Matcher extends MatcherName = MatcherName,
+  Node extends ExpectMember<Matcher> = ExpectMember<Matcher>
+> extends ParsedExpectMember<Matcher, Node> {
+  arguments: TSESTree.CallExpression['arguments'] | null;
+  // isNamed<PossibleName extends Matcher>(
+  //   name: PossibleName,
+  //   ...or: PossibleName[]
+  // ): this is ParsedExpectMatcher<PossibleName>;
+  // isNamed<PossibleName extends Matcher>(
+  //   // name: PossibleName,
+  //   ...names: [PossibleName, ...PossibleName[]]
+  // ): this is ParsedExpectMatcher<PossibleName, Node>;
+}
+
+type BaseParsedModifier<
+  Modifier extends ModifierName = ModifierName
+> = ParsedExpectMember<Modifier>;
+
+type NegatableModifierName = ModifierName.rejects | ModifierName.resolves;
+type NotNegatableModifierName = ModifierName.not;
+
+interface NegatableParsedModifier<
+  Modifier extends NegatableModifierName = NegatableModifierName
+> extends BaseParsedModifier<Modifier> {
+  negation?: ExpectMember<ModifierName.not>;
+}
+
+export interface NotNegatableParsedModifier<
+  Modifier extends NotNegatableModifierName = NotNegatableModifierName
+> extends BaseParsedModifier<Modifier> {
+  negation?: never;
+}
+
+type ParsedExpectModifier =
+  | NotNegatableParsedModifier<NotNegatableModifierName>
+  | NegatableParsedModifier<NegatableModifierName>;
+
+interface Expectation<ExpectNode extends ExpectCall = ValidExpectCall> {
+  expect: ExpectNode;
+  /** @deprecated */
```

Comment:
deprecated? It's introduced in this PR :P

Reply:
hehe that's another one of the things I've been meaning to remove - It was so I could check them all and remove them as part of the clean up XD